compiler/noirc_evaluator/src/ssa/ir/cfg.rs

```diff
@@ -148,6 +149,43 @@ impl ControlFlowGraph {
     pub(crate) fn compute_entry_blocks(&self) -> Vec<BasicBlockId> {
         self.data.keys().filter(|&&block| self.predecessors(block).len() == 0).copied().collect()
     }
+
+    /// Computes the reverse graph of the extended CFG.
+    /// The extended CFG is the CFG with an additional unique exit node (if there is none)
+    /// such that there is a path from every block to the exit node.
+    pub(crate) fn extended_reverse(func: &mut Function) -> Self {
+        let mut cfg = Self::with_function(func);
+        // Exit blocks are the ones having no successor
+        let exit_nodes: Vec<BasicBlockId> =
+            cfg.data.keys().filter(|&&block| cfg.successors(block).len() == 0).copied().collect();
```

Comment:
Just to make sure I understand: both the forward and the reverse CFG in the example only have 1 exit node, however, it's not reachable from every block. That is, `b5` is not an exit node, because it has a successor: itself. 

Not sure if you found a case with multiple exit nodes, but in case you didn't, then perhaps instead of adding a new unique exit node to the DFG and connecting all dead and exit nodes to it, we could connect dead nodes to the in the reverse post order graph to the single exit node of the original? Or would that distort the frontier algorithm?

Reply:
I guess `unreachable` terminators would be the multiple exit node examples.